metricbeat/module/graphite/server/data.go

```diff
@@ -0,0 +1,167 @@
+package server
+
+import (
+	"errors"
+	"math"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+type template struct {
+	Namespace string
+	Delimiter string
+	Parts     []string
+	Tags      map[string]string
+}
+
+type metricProcessor struct {
+	templates       *tree
+	defaultTemplate template
+	sync.RWMutex
+}
+
+func NewMetricProcessor(templates []templateConfig, defaultTemplate templateConfig) metricProcessor {
+
```

Comment:
minor thing, but recently we have been trying to avoid new lines at the start and end of functions: #4670